### PR TITLE
Simplify test262-harness.py

### PR DESCRIPTION
**Pull request by ossy-szeged**
Use regular expressions to parse the YAML headers of the tests
instead of the imported thirdparty YAML parser.

JerryScript-DCO-1.0-Signed-off-by: Csaba Osztrogonác csaba.osztrogonac@h-lab.eu

### CODE REVIEW

File: tools/runners/test262-harness.py [hunk 1]
```diff
@@ -495,19 +225,43 @@ def __init__(self, suite, name, full_path, strict_mode, command_template, module
...
+        if match:
+            for flag in match.group('flags').split(','):
+                if flag.strip():
+                    self.test_record[flag.strip()] = ""
+
+        match = YAML_NEGATIVE_RE.search(header)
+        if match and match.group("phase") and match.group("type"):
+            self.test_record["negative"] = {
+                "phase" : match.group("phase"),
+                "type" : match.group("type")}
```
Comment (rtakacs) [inline tools/runners/test262-harness.py:259]: I think the closing bracket should be in the next line:

```
self.test_record["negative"] = {
    "phase" : match.group("phase"),
    "type" : match.group("type")
}
```

File: tools/runners/test262-harness.py [hunk 2]
```diff
@@ -495,19 +225,43 @@ def __init__(self, suite, name, full_path, strict_mode, command_template, module
...
+        with open(self.full_path, "rb") as file_desc:
+            full_test = file_desc.read()
+
+        match = TEST_RE.search(full_test)
+        header = match.group('header')
+        self.test = match.group('test1') + match.group('test2')
+
+        match = YAML_INCLUDES_RE.search(header)
+        if match:
+            self.test_record["includes"] = [inc.strip() for inc in match.group('includes').split(',') if inc]
```
Comment (rtakacs) [inline tools/runners/test262-harness.py:244]: You should use either apostrophe or double quotation mark characters for string literals.

**Comment by ossy-szeged** [comment]
Based on #4959, it should be merged first.

**Comment by rtakacs** [comment]
LGTM (informal)

**Review by LaszloLango — APPROVED** [review]
LGTM

**Review by robertsipka — APPROVED** [review]
LGTM

**Pull request merged**